File: 算法/数论/欧拉筛.py
def EulerSieve(n):  # n为筛选的范围上限
    prime = []  # 用于存储找到的素数
    is_prime = [True] * (n + 1)  # 初始化标记数组，假设所有数都是素数
    for i in range(2, n + 1):
        if is_prime[i]:  # 如果i是素数
            prime.append(i)  # 将i添加到素数列表
        for p in prime:
            if i * p > n:  # 如果超过了筛选范围，就退出循环
                break
            is_prime[i * p] = False  # 标记合数
            if i % p == 0:  # 关键步骤：保证了每个合数只会被其最小的素因子筛除
                break
    return prime

n = 1000
lis = EulerSieve(n)
for i in lis:
    print(i, end = ' ')

File: 算法模板cpp/欧拉筛.cpp

#include <bits/stdc++.h>
using namespace std;
#define int long long
#define lowbit(x) (x & (-x))
#define N 100

vector<int>prime;

bool is_prime[N];//范围
void eulersiece(int n){//范围
    memset(is_prime,1,sizeof is_prime);
    for (int i = 2; i < n+1; ++i) {
        if (is_prime[i]){
            prime.push_back(i);
        }
        for (int j : prime) {
            if (i*j>n){
                break;
            }
            is_prime[i*j]= false;
            if (i%j==0){
                break;
            }
        }
    }
}

signed main(){
    eulersiece(100);
    for (int i : prime)
    {
        cout<<i<<' ';
    }
    
    return 0;
}